pkg/resourcemanager/webhook/vpainplaceupdates/handler.go

```diff
@@ -0,0 +1,42 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package vpainplaceorrecreateupdatemode
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	vpaautoscalingv1 "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1"
+	"k8s.io/utils/ptr"
+)
+
+// Handler handles admission requests and sets the vpa.spec.updatePolicy.updateMode field in VerticalPodAutoscaler resources.
+type Handler struct {
+	Logger logr.Logger
+}
+
+// Default defaults the update mode of the provided VerticalPodAutoscaler.
+func (h *Handler) Default(_ context.Context, obj runtime.Object) error {
+	vpa, ok := obj.(*vpaautoscalingv1.VerticalPodAutoscaler)
+	if !ok {
+		return fmt.Errorf("expected *vpaautoscalingv1.VerticalPodAutoscaler but got %T", obj)
+	}
+
+	log := h.Logger.WithValues("vpa", vpa.GetName(), "namespace", vpa.GetNamespace())
```

Comment:
What other defaulting webhooks use across the codebase is:

```suggestion
	req, err := admission.RequestFromContext(ctx)
	if err != nil {
		return err
	}

	log := h.Logger.WithValues("vpa", kubernetesutils.ObjectKeyForCreateWebhooks(vpa, req))
```

Examples:
- https://github.com/gardener/gardener/blob/485c25124ea536e4610c627109017dd34d434921/pkg/resourcemanager/webhook/endpointslicehints/handler.go#L31-L36
- https://github.com/gardener/gardener/blob/485c25124ea536e4610c627109017dd34d434921/pkg/resourcemanager/webhook/kubernetesservicehost/handler.go#L33-L38
- https://github.com/gardener/gardener/blob/485c25124ea536e4610c627109017dd34d434921/pkg/resourcemanager/webhook/podtopologyspreadconstraints/handler.go#L66-L71

Reply:
Thanks for the suggestion, updated the handler and reused the `kubernetesutils` when setting up the _structured logging_ `log`.